app/AvailablePlugin/EmailAddressWidget/Model/CoEmailAddressWidget.php

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link          http://www.internet2.edu/comanage COmanage Project
+ * @package       registry-plugin
+ * @since         COmanage Registry v4.1.0
+ * @license       Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ */
+
+App::uses("CoDashboardWidgetBackend", "Model");
+App::uses("CakeEmail", "Network/Email");
+
+class CoEmailAddressWidget extends CoDashboardWidgetBackend {
+  // Define class name for cake
+  public $name = "CoEmailAddressWidget";
+
+  // Add behaviors
+  public $actsAs = array('Containable');
+
+  // Association rules from this model to other models
+  public $belongsTo = array(
+    "CoDashboardWidget"
+  );
+
+  public $hasMany = array(
+    "EmailAddressWidgetVerification" => array(
+      'dependent' => true,
+      'foreignKey' => 'co_email_address_widget_id'
+    )
+  );
+
+  // Validation rules for table elements
+  public $validate = array(
+    'co_dashboard_widget_id' => array(
+      'rule' => 'numeric',
+      'required' => true,
+      'allowEmpty' => false
+    ),
+    'type' => array(
+      'content' => array(
+        'rule' => array('validateExtendedType',
+          array('attribute' => 'EmailAddress.type',
+                'default' => array(EmailAddressEnum::Delivery,
+                  EmailAddressEnum::Forwarding,
+                  EmailAddressEnum::MailingList,
+                  EmailAddressEnum::Official,
+                  EmailAddressEnum::Personal,
+                  EmailAddressEnum::Preferred,
+                  EmailAddressEnum::Recovery))),
+        'required' => false,
+        'allowEmpty' => true
+      )
+    ),
+    'co_message_template_id' => array(
+      'rule' => 'numeric',
+      'required' => false,
+      'allowEmpty' => true
+    )
+  );
+
+  /**
+   * Generate a token for email verification and save it in a new record
+   * along with the email address and email type to be  added.
+   * This is step one of a two-step email verification process.
+   *
+   * @since COmanage Registry v4.1.0
+   * @param string  $emailAddress Email address to be added
+   * @param string  $emailType    Type of email address to be added
+   * @param integer $copersonid   The CO Person to be added the email
+   * @return integer id of the new row
+   */
+  public function generateToken($emailAddress, $emailType, $copersonid) {
+    $token = generateRandomToken(8);
+
+    $fields = array(
+      'email' => $emailAddress,
+      'type' => $emailType,
+      'token' => $token,
+      'co_email_address_widget_id' => $this->id,
+      'co_person_id' => $copersonid
+    );
+
+    $this->EmailAddressWidgetVerification->clear();
+    if(!$this->EmailAddressWidgetVerification->save($fields)) {
+      throw new RuntimeException($this->validationErrors);
+    }
+
+    // Return the token to pass along via email
+    return array(
+      'token' => $token
```

Comment:
Why are we returning an array with a single key? Why don't we just return the token?

Also, either way this isn't consistent with the docblock signature above.